srtcore/congctl.cpp

```diff
@@ -377,34 +377,46 @@ class FileCC: public SrtCongestionControlBase
         else
         {
             m_dCWndSize = m_parent->deliveryRate() / 1000000.0 * (m_parent->RTT() + m_iRCInterval) + 16;
+            HLOGC(mglog.Debug, log << "FileCC: UPD (speed mode) wndsize="
+                << m_dCWndSize << "/" << m_dMaxCWndSize << " RTT = " << m_parent->RTT()
+                << " sndperiod=" << m_dPktSndPeriod << "us. deliverRate = "
+                << m_parent->deliveryRate() << " pkts/s)");
         }
 
-        // No rate increase during Slow Start
         if (!m_bSlowStart)
         {
             if (m_bLoss)
             {
                 m_bLoss = false;
             }
+            // During Slow Start, no rate increase
             else
             {
                 double inc = 0;
-                int64_t B = (int64_t)(m_parent->bandwidth() - 1000000.0 / m_dPktSndPeriod);
-                if ((m_dPktSndPeriod > m_dLastDecPeriod) && ((m_parent->bandwidth() / 9) < B))
-                    B = m_parent->bandwidth() / 9;
+                const int loss_bw = 2 * (1000000 / m_dLastDecPeriod); // 2 times last loss point
+                const int bw_pktps = min(loss_bw, m_parent->bandwidth());
+
+                int64_t B = (int64_t)(bw_pktps - 1000000.0 / m_dPktSndPeriod);
+                if ((m_dPktSndPeriod > m_dLastDecPeriod) && ((bw_pktps / 9) < B))
+                    B = bw_pktps / 9;
                 if (B <= 0)
-                    inc = 1.0 / m_parent->MSS();    // was inc = 0.01; in UDT
+                    inc = 1.0 / m_parent->MSS();    // was inc = 0.01;
```

Comment:
This comment seems to not make sense at all, maybe better remove it if so

Reply:
Deleted.